include/vm/async.h

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+//===-- wasmedge/vm/async.h - Asynchronous Result class definition --------===//
+//
+// Part of the WasmEdge Project.
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file is the definition class of VM class.
+///
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "vm.h"
+#include <future>
+#include <thread>
+
+namespace WasmEdge {
+namespace VM {
+
+/// VM execution flow class
+template <typename T> class Async {
+public:
+  Async() noexcept = default;
+  template <typename... FArgsT, typename... ArgsT>
+  Async(T (VM::*FPtr)(FArgsT...), VM &TargetVM, ArgsT &&...Args)
+      : VMPtr(&TargetVM) {
+    std::promise<T> Promise;
+    Future = Promise.get_future();
+    Thread =
+        std::thread([FPtr, P = std::move(Promise),
+                     Tuple = std::tuple(
+                         &TargetVM, std::forward<ArgsT>(Args)...)]() mutable {
+          std::get<0>(Tuple)->newThread();
+          P.set_value(std::apply(FPtr, Tuple));
+        });
+    Thread.detach();
+  }
+  Async(const Async &) noexcept = delete;
+  Async(Async &&Other) noexcept : Async() { swap(*this, Other); }
+  Async &operator=(const Async &) = delete;
+  Async &operator=(Async &&Other) noexcept {
+    swap(*this, Other);
+    return *this;
+  }
+
+  bool valid() const noexcept { return Future.valid(); }
+
+  T get() { return Future.get(); }
+
+  void wait() const { return Future.wait(); }
```

Comment:
Why do we need to return a void function?

Reply:
Done.